base/services/dhcpcsvc/dhcpcsvc.c

```diff
@@ -47,6 +47,8 @@ DhcpCApiInitialize(LPDWORD Version)
     if (PipeHandle == INVALID_HANDLE_VALUE)
     {
         /* We didn't */
+        HeapFree(GetProcessHeap(), 0, DhcpPipeSD);
+        DhcpPipeSD = NULL;
```

Comment:
You're sure about these heapfree's here? This function is not the one that has managed the pipe creation and its security.